### PR TITLE
Tools for code

### PR DESCRIPTION
**Pull request by vineetshar**
Screenshot 2024-09-09 at 8 59 32 AM
Screenshot 2024-09-09 at 9 03 38 AM
Screenshot 2024-09-09 at 9 43 42 AM

### CODE REVIEW

File: app/celery/celery_app.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+from celery import Celery
+
+from app.core.models import *  # This will import and initialize all models
+
+# Redis configuration
+redishost = os.getenv("REDISHOST", "localhost")
+redisport = int(os.getenv("REDISPORT", 6379))
+redisuser = os.getenv("REDISUSER", "")
+redispassword = os.getenv("REDISPASSWORD", "")
+queue_name = os.getenv("CELERY_QUEUE_NAME", "staging")
```
Comment (dhirenmathur) [inline app/celery/celery_app.py:13]: can we do this in the config.py file that has neo4j creds

File: app/modules/conversations/conversation/conversation_service.py [hunk 2]
```diff
@@ -299,6 +293,7 @@ async def _generate_and_stream_ai_response(
             )
 
         try:
+            print(f"Running agent {conversation.project_ids[0]} with query: {query}")
```
Comment (dhirenmathur) [inline app/modules/conversations/conversation/conversation_service.py]: logger

File: app/modules/github/github_service.py [hunk 3]
```diff
@@ -220,3 +220,55 @@ def get_public_github_repo(repo_name: str):
...
+
+            file_contents = repo.get_contents(file_path)
+
+            if isinstance(file_contents, list):
+                raise HTTPException(
+                    status_code=400, detail="Provided path is a directory, not a file"
+                )
+
+            decoded_content = file_contents.decoded_content.decode("utf-8")
+            lines = decoded_content.split("\n")
```
Comment (dhirenmathur) [inline app/modules/github/github_service.py]: handle non-text files

File: app/celery/celery_app.py (hunk 1 above)
Comment (vineetshar) [reply app/celery/celery_app.py:13]: Ideally no: 
1. They currently get picked from .env which makes sense to me. Adding another layer doesnt.
2. Bigger problem : Right now celery module can be copied as a microservice and doesnt have a lot of dependency. Adding a config layer in between will tightly bind it.

File: app/modules/github/github_service.py [hunk 4]
```diff
@@ -33,8 +31,8 @@ def get_github_repo_details(repo_name: str):
         auth = AppAuth(app_id=app_id, private_key=private_key)
         jwt = auth.create_jwt()
         owner = repo_name.split("/")[0]
-        repo = repo_name.split("/")[1]
-        url = f"https://api.github.com/repos/{owner}/{repo}/installation"
+
+        url = f"https://api.github.com/repos/{owner}/{repo_name.split('/')[1]}/installation"
```
Comment (dhirenmathur) [inline app/modules/github/github_service.py:35]: can we handle local repos in a follwo up PR

File: app/modules/github/github_service.py [hunk 5]
```diff
@@ -204,71 +207,25 @@ def get_branch_list(repo_name: str):
...
-                app_auth = auth.get_installation_auth(response.json()["id"])
-                github = Github(auth=app_auth)
-                repo = github.get_repo(repo_name)
-
-            file_contents = repo.get_contents(file_path)
-
-            if isinstance(file_contents, list):
+                return github, repo
+            except Exception as private_error:
+                logger.error(f"Failed to access private repo: {str(private_error)}")
```
Comment (dhirenmathur) [inline app/modules/github/github_service.py:227]: Follow up PR : can we flip this, do private first and then public

Comment (vineetshar) [reply app/modules/github/github_service.py:227]: sure

File: app/modules/github/github_service.py (hunk 4 above)
Comment (vineetshar) [reply app/modules/github/github_service.py:35]: sure

**Pull request merged**